### PR TITLE
put `zcl_abapgit_default_transport` behind interface

### PR DESCRIPTION
**Pull request by larshp**
🤖 🤖 🤖 

This pull request includes various changes to different classes and interfaces in the codebase. The most important changes include modifying the `zcl_abapgit_default_transport` class by changing the visibility of the `CREATE` method to public and modifying the `restore` and `get` methods to handle default transport requests. Additionally, a new method `get_default_transport` is added to the `zcl_abapgit_factory` class.

Main class changes:

* <a href="diffhunk://#diff-a956e7ec20d767a3f27c0380a3b06f62915122cd58a513f790c7bd3cbb4ebe0cL3-L37">`src/cts/zcl_abapgit_default_transport.clas.abap`</a>: Changed the visibility of the `CREATE` method to public, removed the `set_internal` and `store` methods, and modified the `restore` and `get` methods to handle default transport requests. <a href="diffhunk://#diff-a956e7ec20d767a3f27c0380a3b06f62915122cd58a513f790c7bd3cbb4ebe0cL3-L37">[1]</a> <a href="diffhunk://#diff-a956e7ec20d767a3f27c0380a3b06f62915122cd58a513f790c7bd3cbb4ebe0cL197-L225">[2]</a> <a href="diffhunk://#diff-a956e7ec20d767a3f27c0380a3b06f62915122cd58a513f790c7bd3cbb4ebe0cL136-R161">[3]</a> <a href="diffhunk://#diff-a956e7ec20d767a3f27c0380a3b06f62915122cd58a513f790c7bd3cbb4ebe0cL61-R38">[4]</a> <a href="diffhunk://#diff-a956e7ec20d767a3f27c0380a3b06f62915122cd58a513f790c7bd3cbb4ebe0cL149-R174">[5]</a> <a href="diffhunk://#diff-a956e7ec20d767a3f27c0380a3b06f62915122cd58a513f790c7bd3cbb4ebe0cL91-R149">[6]</a>
* <a href="diffhunk://#diff-b952028b68ab9dddec340289a6820a56b803d6857b617bec1c8bb4af7c303f5fR244-R253">`src/zcl_abapgit_factory.clas.abap`</a>: Added a new method `get_default_transport` and a new class data `gi_default_transport` of type `zif_abapgit_default_transport`. <a href="diffhunk://#diff-b952028b68ab9dddec340289a6820a56b803d6857b617bec1c8bb4af7c303f5fR244-R253">[1]</a> <a href="diffhunk://#diff-b952028b68ab9dddec340289a6820a56b803d6857b617bec1c8bb4af7c303f5fR85">[2]</a>
* <a href="diffhunk://#diff-d3f4dd2eeba5e6e323ade60149b325d8ec348e585278414107e0bcb247a63560L329-R329">`src/repo/zcl_abapgit_repo.clas.abap`</a>: Replaced the `reset` method in the `zcl_abapgit_repo` class with the `reset` method in the `zcl_abapgit_factory` class.
* <a href="diffhunk://#diff-73ff03c5b84288687399d8d25cdd89ef14c62735391ffaf91d3fdbb3380e5896L635-R635">`src/objects/zcl_abapgit_objects.clas.abap`</a>: Replaced the `set` and `reset` methods of `zcl_abapgit_default_transport` in the `zcl_abapgit_objects` class with the corresponding methods of `zcl_abapgit_factory`. Removed a line of code from the `update_package_tree` method. <a href="diffhunk://#diff-73ff03c5b84288687399d8d25cdd89ef14c62735391ffaf91d3fdbb3380e5896L635-R635">[1]</a> <a href="diffhunk://#diff-73ff03c5b84288687399d8d25cdd89ef14c62735391ffaf91d3fdbb3380e5896L527-R527">[2]</a> <a href="diffhunk://#diff-73ff03c5b84288687399d8d25cdd89ef14c62735391ffaf91d3fdbb3380e5896L514-R514">[3]</a> <a href="diffhunk://#diff-73ff03c5b84288687399d8d25cdd89ef14c62735391ffaf91d3fdbb3380e5896L577-R577">[4]</a> <a href="diffhunk://#diff-73ff03c5b84288687399d8d25cdd89ef14c62735391ffaf91d3fdbb3380e5896L812-R812">[5]</a>

Interface changes:

* <a href="diffhunk://#diff-48f67e6069500d4d65afd8769413e0b5f1a48f2040e045e6acabeecbbbc9be7cR1-R15">`src/cts/zif_abapgit_default_transport.intf.xml`</a>: Added a new XML file to define the `zif_abapgit_default_transport` interface.
* <a href="diffhunk://#diff-73d70e199bd13cbb811831c7653892133253801a059f2e7898d1187483b0946eR1-R24">`src/cts/zif_abapgit_default_transport.intf.abap`</a>: Added a new interface `zif_abapgit_default_transport` with three methods (`set`, `reset`, and `get`) to handle transport request numbers and raise exceptions if errors occur.

### CODE REVIEW

File: src/objects/zcl_abapgit_object_sod1.clas.abap [hunk 1]
```diff
@@ -350,7 +350,7 @@ CLASS zcl_abapgit_object_sod1 IMPLEMENTATION.
         lo_factory = get_wb_object_operator( is_object_type = ls_object_type
                                              iv_object_key  = lv_object_key ).
 
-        lv_transport_request = zcl_abapgit_default_transport=>get_instance( )->get( )-ordernum.
+        lv_transport_request = zcl_abapgit_factory=>get_default_transport( )->get( )-ordernum.
```
Comment (mbtools) [inline src/objects/zcl_abapgit_object_sod1.clas.abap:353]: this should actually use `iv_transport`

File: src/objects/zcl_abapgit_object_sod2.clas.abap [hunk 2]
```diff
@@ -349,7 +349,7 @@ CLASS zcl_abapgit_object_sod2 IMPLEMENTATION.
         lo_factory = get_wb_object_operator( is_object_type = ls_object_type
                                              iv_object_key  = lv_object_key ).
 
-        lv_transport_request = zcl_abapgit_default_transport=>get_instance( )->get( )-ordernum.
+        lv_transport_request = zcl_abapgit_factory=>get_default_transport( )->get( )-ordernum.
```
Comment (mbtools) [inline src/objects/zcl_abapgit_object_sod2.clas.abap:352]: `iv_transport`

**Review by mbtools — CHANGES_REQUESTED** [review]
small change for `sod1/2`

**Comment by mbtools** [comment]
co-pilot loves some detail 😀

**Comment by larshp** [comment]
> co-pilot loves some detail 😀

yea, its not super useful IMHO

File: src/objects/zcl_abapgit_object_sod1.clas.abap (hunk 1 above)
Comment (larshp) [reply src/objects/zcl_abapgit_object_sod1.clas.abap]: fixed

File: src/objects/zcl_abapgit_object_sod2.clas.abap (hunk 2 above)
Comment (larshp) [reply src/objects/zcl_abapgit_object_sod2.clas.abap]: fixed

**Review by mbtools — APPROVED** [review]
👍

**Pull request merged**